File: patterns/event-debouncer/rules-js/event-debouncer.js
let debounceTimer = null;

rules.JSRule({
  name: "Debounced Motion Sensor",
  triggers: [triggers.ItemStateChangeTrigger("motionSensor")],
  execute: (event) => {
    if (debounceTimer === null) {
      console.log("Bewegung erkannt - Licht einschalten");
      items.getItem("light").sendCommand("ON");

      debounceTimer = actions.ScriptExecution.createTimer(
        time.ZonedDateTime.now().plusSeconds(2),
        () => {
          debounceTimer = null;
          console.log("Debounce beendet - neue Events moeglich");
        }
      );
    } else {
      console.log("Event ignoriert - Timer laeuft noch");
    }
  }
});

File: patterns/event-debouncer/rules-python/event_debouncer.py
import threading

from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger

debounce_timer = None


@rule(triggers=[ItemStateChangeTrigger("motionSensor")])
class DebouncedMotionSensor:
    def execute(self, module, input):
        global debounce_timer

        if debounce_timer is None:
            self.logger.info("Bewegung erkannt - Licht einschalten")
            Registry.getItem("light").sendCommand("ON")

            def reset_timer():
                global debounce_timer
                debounce_timer = None
                self.logger.info("Debounce beendet - neue Events moeglich")

            debounce_timer = threading.Timer(2, reset_timer)
            debounce_timer.start()
        else:
            self.logger.info("Event ignoriert - Timer laeuft noch")
